src/common/containers/build-details.js

```diff
@@ -5,18 +5,37 @@ import Helmet from 'react-helmet';
 import BuildRow from '../components/build-row';
 import BuildLog from '../components/build-log';
 import { Message } from '../components/forms';
-import { fetchSnap } from '../actions/snap-builds';
+
+import { getGitHubRepoUrl } from '../helpers/github-url';
+import { fetchBuilds, fetchSnap } from '../actions/snap-builds';
 
 import styles from './container.css';
 
 class BuildDetails extends Component {
 
+  fetchData({ snapLink, repository }) {
+    if (snapLink) {
+      this.props.dispatch(fetchBuilds(snapLink));
+    } else {
+      this.props.dispatch(fetchSnap(repository.url));
+    }
+  }
+
   componentWillMount() {
-    this.props.dispatch(fetchSnap(this.props.fullName));
+    this.fetchData(this.props);
```

Comment:
Properly fetching snap and build data when user enters build details directly via URL.